Documentation/format/tombstones.md

```diff
@@ -0,0 +1,55 @@
+# Tombstones Disk Format
+
+The following describes the format of a tombstones file, which is the directory of a block.
+
+The last 8 bytes specifies the offset to the start of Stones section.
+The stones section is 0 padded to a multiple of 4 for fast scans.
+
+```
+┌────────────────────────────┬─────────────────────┐
+│ magic(0x130BA30) <4b>      │ version(1) <1 byte> │
+├────────────────────────────┴─────────────────────┤
+│ ┌──────────────────────────────────────────────┐ │
+│ │                   Ranges 1                   │ │
+│ ├──────────────────────────────────────────────┤ │
+│ │                      ...                     │ │
+│ ├──────────────────────────────────────────────┤ │
+│ │                   Ranges N                   │ │
+│ ├──────────────────────────────────────────────┤ │
+│ │                    Stones                    │ │
+│ ├──────────────────────────────────────────────┤ │
+│ │             Ref(stones start)<8b>            │ │
+│ └──────────────────────────────────────────────┘ │
+└──────────────────────────────────────────────────┘
```

Comment:
What is advantage of having a level of indirection from stone to ranges?
Right now (and probably for ever), we are scanning all stones and reading them into memory. In the general case, there's probably just a single deletion range per series (or a few). The total size of ranges for a stone are still tiny, so separating them away from the stone itself just adds complexity to the format and related code.

It seems fine here to just make the file a sequence of `<series_ref>, <#ranges>, <range1>, <range2>, ...`

The reason why we chose indirection elsewhere is usually to apply deduplication (symbol table in the index) or to co-locate frequently accessed, small items better by moving larger contents belonging to them elsewhere (series in the index vs the chunks). Here neither is really a concern.

Probably fine to just CRC all entries once at the end. It will generally be in the single digit MB range even when deleting 200k series or so. At that scale, it will likely be applied via compaction directly because we can save a lot of space by removing data.
If a tombstone file is small enough to leave it around, a CRC32 is definitely sufficient.